wagtail/images/rich_text.py

```diff
@@ -39,16 +65,55 @@ def expand_db_attributes(attrs):
         return image_format.image_to_editor_html(image, attrs.get('alt', ''))
 
 
-def image_embedtype_handler(attrs):
+EditorHTMLImageConversionRule = [
+    editor_html.EmbedTypeRule('image', ImageEmbedHandler)
+]
+
+
+# draft.js / contentstate conversion
+
+def image_entity(props):
     """
-    Given a dict of attributes from the <embed> tag, return the real HTML
-    representation for use on the front-end.
+    Helper to construct elements of the form
+    <embed alt="Right-aligned image" embedtype="image" format="right" id="1"/>
+    when converting from contentstate data
     """
-    Image = get_image_model()
-    try:
-        image = Image.objects.get(id=attrs['id'])
-    except Image.DoesNotExist:
-        return "<img>"
+    return DOM.create_element('embed', {
+        'embedtype': 'image',
+        'format': props.get('format'),
+        'id': props.get('id'),
+        'alt': props.get('alt'),
+    })
 
-    image_format = get_image_format(attrs['format'])
-    return image_format.image_to_html(image, attrs.get('alt', ''))
+
+class ImageElementHandler(AtomicBlockEntityElementHandler):
+    """
+    Rule for building an image entity when converting from database representation
+    to contentstate
+    """
+    def create_entity(self, name, attrs, state, contentstate):
+        Image = get_image_model()
+        try:
+            image = Image.objects.get(id=attrs['id'])
+            image_format = get_image_format(attrs['format'])
+            rendition = get_rendition_or_not_found(image, image_format.filter_spec)
+            src = rendition.url
+        except Image.DoesNotExist:
+            src = ''
```

Comment:
What should happen on the front-end if there is no `src`?

Reply:
If I'm not mistaken, we can't edit an image block yet? So like my answer for page/document link, I believe we should remove the block?